d2common/math.go

```diff
@@ -89,3 +91,36 @@ func GetRadiansBetween(p1X, p1Y, p2X, p2Y float64) float64 {
 func AlmostEqual(a, b, threshold float64) bool {
 	return math.Abs(a-b) <= threshold
 }
+
+// Return the new adjusted value, as well as any remaining amount after the max
+func AdjustWithRemainder(sourceValue, adjustment, targetvalue float64) (newValue, remainder float64) {
+	if adjustment == 0 || math.Abs(adjustment) < 0.000001 {
+		return sourceValue, 0
+	}
+	adjustNegative := adjustment < 0.0
+	maxNegative := targetvalue-sourceValue < 0.0
+	if adjustNegative != maxNegative {
+		// FIXME: This shouldn't happen but it happens all the time..
+		return sourceValue, 0
+		//panic("Cannot move towards the opposite direction...")
+	}
+
+	finalValue := sourceValue + adjustment
+	if !adjustNegative {
+		if finalValue > targetvalue {
+			diff := finalValue - targetvalue //  RoundToDecial(finalValue-targetvalue, 6)
+			return targetvalue, diff
+		}
+		return finalValue, 0
+	}
+
+	if finalValue < targetvalue {
+		return targetvalue, RoundToDecial(finalValue-targetvalue, 6)
+	}
+	return finalValue, 0
+}
+
+func RoundToDecial(f float64, d int) float64 {
```

Comment:
😆

Reply:
The struggle was real.

Reply:
I know you have GoLand.  Fix the squiggly lines!